### PR TITLE
Feature unit tests

### PR DESCRIPTION
**Pull request by AussieKing**
Added 2x other unit tests for the Services!

### CODE REVIEW

**Comment by AussieKing** [comment]
have changed the lifetime of `TrolleyService` to only be Scoped (in Program.cs , is the `builder.Services.AddScoped<ITrolleyService, TrolleyService>();` )

Also checking with some tests `TrolleyService_ShouldProvideSeparateInstancesForDifferentScopes` to make sure the instances are independent of each other.

Is this what you meant as solution @ssu8 ?

I think we could also now merge the `feature/trolley-api` into the `feature-unit-tests` ...?

Thank you again for today, was a great explanation!

File: tests/BeerAPI.Tests/Controllers/BeersControllersTests.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        public void GetBeerById_NonExistingId_ReturnsNotFoundResult()
+        {
+            // Arrange
+            _mockBeerService.Setup(service => service.GetBeerById(It.IsAny<int>())).Returns(default(Beer?));
+
+            // Act
+            var result = _controller.GetBeerById(999); // this ID doesn't exist
+
+            // Assert
+            Assert.IsType<NotFoundResult>(result);
```
Comment (ssu8) [inline tests/BeerAPI.Tests/Controllers/BeersControllersTests.cs:34]: Checking for `NotFoundResult` is a good start, but have a think about what that tells future engineers. Consider the  asserts below. What thoughts will a future engineer reading these different asserts have?
```suggestion
            var statusCodeResult = Assert.IsAssignableFrom<IStatusCodeActionResult>(result);
            Assert.Equal(404, statusCodeResult.StatusCode);
```

File: tests/BeerAPI.Tests/Services/TrolleyServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+        public TrolleyServiceTests()
+        {
+            _trolleyService = new TrolleyService();
+        }
+
+        private void ResetTrolleyServiceState()
+        {
+            _trolleyService = new TrolleyService(); // adding this to re-set the state of the service before each test
+        }
```
Comment (ssu8) [inline tests/BeerAPI.Tests/Services/TrolleyServiceTests.cs:20]: You don't need this. xUnit will reset the state before each unit test executes.

File: tests/BeerAPI.Tests/Controllers/BeersControllersTests.cs (hunk 1 above)
Comment (AussieKing) [reply tests/BeerAPI.Tests/Controllers/BeersControllersTests.cs:34]: makes sense!
`IsAssignableFrom` is less rigid than `IsType` (could be any type that implements the `IStatusCodeActionResult`) ad the checking of `statusCodeResult.StatusCode` being 404 means that we are telling future engineers what the status SHOULD be ! Thanks @ssu8 !

**Pull request merged**